code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -470,30 +485,73 @@ func getBlobbersForRequest(request newAllocationRequest, edb *event.EventDb, bal
 //  400:
 func (srh *StorageRestHandler) getCollectedReward(w http.ResponseWriter, r *http.Request) {
 	var (
-		startBlock, _ = strconv.Atoi(r.URL.Query().Get("start_block"))
-		endBlock, _   = strconv.Atoi(r.URL.Query().Get("end_block"))
-		clientID      = r.URL.Query().Get("client_id")
+		startBlock, _    = strconv.Atoi(r.URL.Query().Get("start-block"))
+		endBlock, _      = strconv.Atoi(r.URL.Query().Get("end-block"))
+		clientID         = r.URL.Query().Get("client-id")
+		startDateString  = r.URL.Query().Get("start-date")
+		endDateString	 = r.URL.Query().Get("end-date")
+		dataPointsString = r.URL.Query().Get("data-points") 
 	)
 
+	var dataPoints int64
+	dataPoints, err := strconv.ParseInt(dataPointsString, 10, 64)
+	if err != nil {
+		dataPoints = 1
+	}
+
+
 	query := event.RewardQuery{
 		StartBlock: startBlock,
 		EndBlock:   endBlock,
 		ClientID:   clientID,
+		DataPoints: dataPoints,
 	}
 	edb := srh.GetQueryStateContext().GetEventDB()
 	if edb == nil {
 		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
 		return
 	}
-	collectedReward, err := edb.GetRewardClaimedTotal(query)
-	if err != nil {
-		common.Respond(w, r, 0, common.NewErrInternal("can't get rewards claimed", err.Error()))
-		return
+
+
+	if startBlock > 0 && endBlock > 0 {
+		collectedReward, err := edb.GetRewardClaimedTotalBetweenBlocks(query)
+		if err != nil {
+			common.Respond(w, r, 0, common.NewErrInternal("can't get rewards claimed", err.Error()))
+			return
+		}
+		common.Respond(w, r, map[string][]int64{
+			"collected_reward": {collectedReward},
+		}, nil)
 	}
 
-	common.Respond(w, r, map[string]int64{
-		"collected_reward": collectedReward,
-	}, nil)
+	if startDateString != "" && endDateString != "" {
```

Comment:
Umm, my idea was to have ability to resolve dates to blockNum and if both blocks and dates provided we could ignore dates??

Reply:
The http GET  url query params works in `AND` condition, such as `?start-block=10&end-block=100&from-date=xx&end-date=xxx`, people would think the result are merged, otherwise we'd better comment about it in swagger doc.

Reply:
Yeah that makes sense. I will add in the swagger docs.